include/mbedtls/platform_util.h

```diff
@@ -41,6 +41,68 @@
 extern "C" {
 #endif
 
+#if defined(MBEDTLS_CHECK_PARAMS)
+
+#if !defined(MBEDTLS_PARAM_FAILED)
+
+#define MBEDTLS_PARAM_FAILED( cond, file, line ) \
+    mbedtls_param_failed( cond, file, line )
+
+/**
+ * \brief       User supplied callback function for parameter validation failure.
+ *
+ *              When the MBEDTLS_CHECK_PARAMS option is enabled, the library
+ *              provides additional validation of all input parameters to
+ *              confirm that they conform to what the interface can accept.
+ *              For example - NULL paramater checks.
+ *
+ *              These checks are designed to check programmatic issues in the
+ *              application software using Mbed TLS, or catch other runtime
+ *              errors which may be due to issues in the application software.
+ *
+ *              This function will be called unless an alternative function is
+ *              defined through the MBEDTLS_PARAM_FAILURE function.
+ *
+ *              This function can return, and the operation will be aborted, or
+ *              alternatively, through use of setjmp()/longjmp() can resume
+ *              execution in the application code.
+ */
+void mbedtls_param_failed( const char* failure_condition, const char* file,
```

Comment:
Perhaps I am missing something, but I do not think this is going to work very well. The problem is that files that include platform_util.h will not find the declaration for mbedtls_param_failed() if you conditionally remove this line based on the definition for MBEDTLS_PARAM_FAILED. This is the reason why MBEDTLS_*_ALT macros generally allow you to create a new .h files. Alternatively, the mbedtls_platform_zeroize() is always declared unconditionally.